homeassistant/components/snooz/__init__.py

```diff
@@ -0,0 +1,52 @@
+"""The Snooz component."""
+from __future__ import annotations
+
+from pysnooz.device import SnoozDevice
+
+from homeassistant.components.bluetooth import async_ble_device_from_address
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, CONF_TOKEN, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+from .models import SnoozConfigurationData
+
+PLATFORMS: list[Platform] = [Platform.FAN]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Snooz device from a config entry."""
+    address: str = entry.data[CONF_ADDRESS]
+    token: str = entry.data[CONF_TOKEN]
+
+    if not (ble_device := async_ble_device_from_address(hass, address)):
+        raise ConfigEntryNotReady(
+            f"Could not find Snooz with address {address}. Try power cycling the device"
+        )
+
+    device = SnoozDevice(ble_device, token, hass.loop)
```

Comment:
Can you avoid passing the loop here and use `asyncio.get_running_loop()` instead?

Reply:
Pass `asyncio.get_running_loop()` instead of `hass.loop`, or remove the parameter and call `asyncio.get_running_loop()` from the library?

Reply:
Remove the param

In general python is moving away from passing around loop so we are trying to do the same